src/png.imageio/pngoutput.cpp

```diff
@@ -78,13 +80,27 @@ class PNGOutput final : public ImageOutput {
         m_convert_alpha = true;
         m_gamma = 1.0;
         m_pngtext.clear ();
+        m_io = nullptr;
     }
 
     // Add a parameter to the output
     bool put_parameter (const std::string &name, TypeDesc type,
                         const void *data);
 
-    void finish_image ();
+    // Callback for PNG that appends to in-memory buffer instead of writing
+    // to a file.
+    static void PngWriteCallback (png_structp  png_ptr, png_bytep data, png_size_t length) {
+        PNGOutput *p = (PNGOutput *)png_get_io_ptr(png_ptr);
+        if (p->m_io)
```

Comment:
Why not pass `m_io` as the user pointer?

It can't be `nullptr` here right?

Reply:
You're right, I can cut out the middleman here.

Reply:
Updated based on this suggestion.